### ISSUE TITLE
Loading graph from sparse matrix creates unwanted edges if matrix antires are 0.

### ISSUE THREAD

**Issue by tobiasschumacher** [networkx/networkx#4472]
Hi,

when trying to simulate [Kronecker graph models](https://cs.stanford.edu/people/jure/pubs/kronecker-jmlr10.pdf) with networkx, I ran into the following issue:

Computing the Kronecker product of two sparse (adjacency) matrices in scipy naturally yields sparse matrices which will also contain some "present" entries which are 0 (other than entries that are omitted in the sparse representation due to being zero). 

When I then initialize a networkx graph from such a sparse matrix, these zero entries will then be interpreted as edges with weight zero, so in particular some edges will be added which should not be contained in the graph.

In consequence, I think the initialization of networks from sparse matrices should be updated, so that present zeros in the sparse adjacency matrix should also be interpreted as non-edges.

Here is an example of performing the same operations with numpy arrays and scipy sparse matrices, yielding diffrent graphs:

```import numpy as np
import networkx as nx
from scipy import sparse

graph = nx.Graph()
graph.add_edges_from([(0,0),(1,1),(2,2),(1,0),(0,1),(1,2),(2,1)])
matrix = nx.to_numpy_matrix(graph)

M = sparse.csr_matrix(matrix)
result_sparse1 = sparse.kron(M,M)
result_np1 = np.kron(matrix,matrix)

graph_sparse = nx.Graph(result_sparse)
graph_np = nx.Graph(result_np)

# This will print False:
print(nx.is_isomorphic(graph_sparse, graph_np))
```

**Comment by rossbar** [comment]
> In consequence, I think the initialization of networks from sparse matrices should be updated, so that present zeros in the sparse adjacency matrix should also be interpreted as non-edges.

It's possible that users rely on the current behavior though. If a user has a sparse matrix and has explicitly chosen to include 0's, it would be inappropriate for the conversion function to drop those values.

In this case, this feels like something better handled in the sparse-computation part, i.e. `sparse.kron` should not (or have an option not to) include 0's in the output (keeping zeros in the sparse output seems to reduce the utility of the sparsification in the first place). Of course, you could always sparsify after the `kron` operation, which gives you the result you want, e.g.
```
result_sparse = sparse.csr_matrix(np.kron(matrix, matrix))
```
but then you have a "dense" step in the calculation (again, this feels like a limitation of sparse.kron).

**Comment by dschult** [comment]
I agree that this should be fixed in `sparse.kron`.  It is legal and desirable in some strange cases to have a zero weight edge.  If you need a workaround, you can remove the edges after-the-fact easily with:

```python
G.remove_edges_from((u, v) for u, v, wt in G.edges.data(weight) if wt == 0)
```

**Comment by rossbar** [comment]
I'm going to close this one as "won't fix", as I think we agree that the source of the problem originates outside of networkx's sparse conversion functions.
